pkg/queue/sharedmain/main.go

```diff
@@ -110,22 +104,45 @@ type config struct {
 	// Concurrency State Endpoint configuration
 	ConcurrencyStateEndpoint  string `split_words:"true"` // optional
 	ConcurrencyStateTokenPath string `split_words:"true"` // optional
+
+	Env
+}
+
+type Env struct {
+	ServingNamespace     string `split_words:"true" required:"true"`
+	ServingRevision      string `split_words:"true" required:"true"`
+	ServingConfiguration string `split_words:"true" required:"true"`
+	ServingPodIP         string `split_words:"true" required:"true"`
+	ServingPod           string `split_words:"true" required:"true"`
+	ServingService       string `split_words:"true"` // optional
```

Comment:
a couple of minor nits: 
* maybe don't rename the `config` struct ?
* perhaps a more descriptive name than `Env` (`PodDetails`? Caveat here is that I'm terrible at naming...)
* I'd put `Env` under the metrics config section (i.e. where the old fields were removed)

Reply:
Note that Env struct is identical in content to:
type RequestLogRevision struct from https://pkg.go.dev/knative.dev/serving/pkg/http#RequestLogRevision which is also being used later in main.go, How about naming it "Revision"